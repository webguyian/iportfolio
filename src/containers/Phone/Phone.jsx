import React from 'react';
import PropTypes from 'prop-types';

import Avatar from 'components/Avatar/Avatar';
import Flip from 'components/Flip/Flip';
import Icon from 'components/Icon/Icon';
import Link from 'components/Link/Link';
import Text from 'components/Text/Text';

import {
  COORDINATES_PHILADELPHIA,
  STATIC_MAP_API
} from 'modules/map/constants';

const Phone = () => {
  const baseClass = 'phone-app';
  const imageClass = `${baseClass}-image`;
  const actionsClass = `${baseClass}-actions`;
  const infoClass = `${baseClass}-info`;
  const actions = [
    {
      to: '/messages',
      icon: 'comment',
      label: 'message'
    },
    {
      to: '#call',
      icon: 'phone-alt',
      label: 'call'
    },
    {
      to: '/mail',
      icon: 'envelope',
      label: 'mail'
    }
  ];
  const links = [
    {
      label: 'Email',
      to: 'mailto:hello@webguyian.com',
      value: 'hello@webguyian.com'
    },
    {
      label: 'Website',
      to: 'https://webguyian.com',
      value: 'webguyian.com'
    },
    {
      label: 'Github',
      to: 'https://github.com/webguyian',
      value: 'github.com/webguyian'
    }
  ];

  return (
    <div className={baseClass}>
      <header>
        <figure className={imageClass}>
          <Flip back={<Avatar alt="Ian Mac" text="IM" />}>
            <Avatar
              alt="Ian Mac"
              src="https://github.com/webguyian.png?size=150"
            />
          </Flip>
          <figcaption className={`${imageClass}-caption`}>
            <Text element="h1">Ian Mac</Text>
          </figcaption>
        </figure>
        <ul className={actionsClass}>
          {actions.map(action => (
            <li key={action.icon}>
              <Link to={action.to} icon={action.icon} iconSize="lg" withLabel>
                {action.label}
              </Link>
            </li>
          ))}
        </ul>
      </header>
      <div className={infoClass}>
        {links.map(link => (
          <div key={link.label} className={`${infoClass}-row`}>
            <Text className={`${infoClass}-label`} element="h2">
              {link.label}
            </Text>
            <Link
              className={`${infoClass}-value`}
              to={link.to}
              data-hover={link.value}
              external
            >
              {link.value}
            </Link>
          </div>
        ))}
        <div className={`${infoClass}-row`}>
          <Text className={`${infoClass}-label`} element="h2">
            Location
          </Text>
          <Link
            to="/map"
            className={`${infoClass}-map-link`}
            state={{
              theme: 'light',
              center: COORDINATES_PHILADELPHIA
            }}
          >
            <Text className={`${infoClass}-map-link-text`}>
              <Icon name="map-marker-alt" /> View Map
            </Text>
            <img alt="Philadelphia, PA" src={STATIC_MAP_API} />
          </Link>
        </div>
      </div>
    </div>
  );
};

Phone.propTypes = {
  className: PropTypes.string
};

export default Phone;
