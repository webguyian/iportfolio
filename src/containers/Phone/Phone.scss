.phone-app {
  @include stretch-column-nowrap;
  align-items: center;
  padding: $gutter;
  background-color: $black-dark;

  &-image {
    margin: $gutter-m auto;
    text-align: center;

    &-caption {
      margin-top: $gutter-s;

      > .ui-text {
        margin: 0;
        font-size: $font-size-l;
      }
    }
  }

  &-actions {
    @include unstyled-list;
    display: flex;

    .ui-link--with-icon,
    .ui-icon {
      display: flex;
      flex-flow: column wrap;
      align-items: center;
      justify-content: center;
    }

    .ui-link--with-icon {
      min-width: 55px;
      margin: 0 $gutter;
      transition: color 0.4s ease-in;
      color: $white;

      > .ui-icon {
        @include circle(44px);
        transition: background-color 0.4s ease-in;
        background-color: $blue;
        line-height: 1.65;
      }

      > .ui-link-label {
        margin: $gutter-s 0 0;
        transition: color 0.4s ease-in;
        color: $blue;
        font-size: $font-size-s;
      }

      &:hover,
      &:focus {
        color: $blue;

        > .ui-icon {
          background-color: $white;
        }

        > .ui-link-label {
          color: $white;
        }
      }
    }
  }

  &-info {
    width: 100%;
    margin-top: $gutter;

    &-row {
      padding: $gutter $gutter-s;
      border-top: 1px solid $gray-medium;
    }

    &-label {
      margin-bottom: $gutter-xs;
    }

    &-value {
      position: relative;
      transition: color 0.4s ease-in;
      color: $blue;

      &::before {
        position: absolute;
        left: 0;
        width: 0;
        transition: width 0.4s;
        color: $white;
        white-space: nowrap;
        content: attr(data-hover);
        overflow: hidden;
      }

      &:hover,
      &:focus {
        &::before {
          width: 100%;
        }
      }
    }

    &-map-link {
      display: block;
      position: relative;
      max-height: 100px;

      &-text {
        @include type-body($font-size-l);
        display: flex;
        position: absolute;
        align-items: center;
        justify-content: center;
        width: 100%;
        height: 100px;
        transition: opacity 0.8s ease-in;
        text-align: center;
        opacity: 0;

        > .ui-icon {
          margin-right: $gutter-s;
        }
      }

      &::before {
        position: absolute;
        left: 0;
        width: 0;
        height: 100%;
        transition: width 0.8s;
        content: "";
      }

      &:hover,
      &:focus {
        .phone-app-info-map-link-text {
          opacity: 1;
        }

        &::before {
          width: 330px;
          background-color: rgba($black, 0.75);
          color: $red;
        }
      }
    }
  }
}
