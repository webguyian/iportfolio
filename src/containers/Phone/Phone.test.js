import React from 'react';
import { create } from 'react-test-renderer';

import Phone from './Phone';

describe('<Phone />', () => {
  it('renders correctly', () => {
    const component = create(<Phone />);

    expect(component).toMatchSnapshot();
  });
});
