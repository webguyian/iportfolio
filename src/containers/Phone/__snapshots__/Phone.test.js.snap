// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Phone /> renders correctly 1`] = `
<div
  className="phone-app"
>
  <header>
    <figure
      className="phone-app-image"
    >
      <div
        className="ui-flip-container"
      >
        <div
          className="ui-flip"
        >
          <div
            className="ui-flip-front"
          >
            <div
              className="ui-avatar"
            >
              <img
                alt="Ian Mac"
                className="ui-avatar-image"
                src="https://github.com/webguyian.png?size=150"
              />
            </div>
          </div>
          <div
            className="ui-flip-back"
          >
            <div
              className="ui-avatar"
            >
              <span
                className="ui-text ui-text--body ui-text--light ui-avatar-text"
              >
                IM
              </span>
            </div>
          </div>
        </div>
      </div>
      <figcaption
        className="phone-app-image-caption"
      >
        <h1
          className="ui-text ui-text--body"
        >
          Ian Mac
        </h1>
      </figcaption>
    </figure>
    <ul
      className="phone-app-actions"
    >
      <li>
        <Link
          icon="comment"
          iconSize="lg"
          to="#message"
          withLabel={true}
        >
          message
        </Link>
      </li>
      <li>
        <Link
          icon="phone-alt"
          iconSize="lg"
          to="#call"
          withLabel={true}
        >
          call
        </Link>
      </li>
      <li>
        <Link
          icon="envelope"
          iconSize="lg"
          to="/mail"
          withLabel={true}
        >
          mail
        </Link>
      </li>
    </ul>
  </header>
  <div
    className="phone-app-info"
  >
    <div
      className="phone-app-info-row"
    >
      <h2
        className="ui-text ui-text--body phone-app-info-label"
      >
        Email
      </h2>
      <Link
        className="phone-app-info-value"
        data-hover="hello@webguyian.com"
        external={true}
        to="mailto:hello@webguyian.com"
      >
        hello@webguyian.com
      </Link>
    </div>
    <div
      className="phone-app-info-row"
    >
      <h2
        className="ui-text ui-text--body phone-app-info-label"
      >
        Website
      </h2>
      <Link
        className="phone-app-info-value"
        data-hover="webguyian.com"
        external={true}
        to="https://webguyian.com"
      >
        webguyian.com
      </Link>
    </div>
    <div
      className="phone-app-info-row"
    >
      <h2
        className="ui-text ui-text--body phone-app-info-label"
      >
        Github
      </h2>
      <Link
        className="phone-app-info-value"
        data-hover="github.com/webguyian"
        external={true}
        to="https://github.com/webguyian"
      >
        github.com/webguyian
      </Link>
    </div>
    <div
      className="phone-app-info-row"
    >
      <h2
        className="ui-text ui-text--body phone-app-info-label"
      >
        Location
      </h2>
      <Link
        className="phone-app-info-map-link"
        state={
          Object {
            "center": Object {
              "lat": 39.952583,
              "lng": -75.165222,
            },
            "theme": "light",
          }
        }
        to="/map"
      >
        <span
          className="ui-text ui-text--body phone-app-info-map-link-text"
        >
          <i
            className="ui-icon ui-icon-map-marker-alt fas fa-map-marker-alt"
          />
           View Map
        </span>
        <img
          alt="Philadelphia, PA"
          src="https://maps.googleapis.com/maps/api/staticmap?center=Philadelphia,PA&zoom=13&size=330x100&maptype=roadmap&key=AIzaSyAc4ldBsk1nC8zCTpCgjP8la4m_uYjmtQM&style=feature:administrative.land_parcel%7Celement:labels%7Cvisibility:off&style=feature:poi%7Celement:labels.text%7Cvisibility:off&style=feature:poi.business%7Cvisibility:off&style=feature:poi.park%7Celement:labels.text%7Cvisibility:off&style=feature:road.arterial%7Celement:labels%7Cvisibility:off&style=feature:road.highway%7Celement:labels%7Cvisibility:off&style=feature:road.local%7Cvisibility:off&style=feature:road.local%7Celement:labels%7Cvisibility:off"
        />
      </Link>
    </div>
  </div>
</div>
`;
